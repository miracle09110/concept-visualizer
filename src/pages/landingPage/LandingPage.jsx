import React from "react";
import HeaderView from "./header/header.view";
import ExploreController from "./explore/Explore.controller";
import { Outlet } from "react-router";
import Navbar from "../../components/navbar/Navbar";

const LandingPage = () => {
  return (
    <>
      <Navbar />
      {/* <ExploreController /> */}
      <Outlet />
    </>
  );
};

export default LandingPage;
