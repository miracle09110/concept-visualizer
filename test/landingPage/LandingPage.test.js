import renderer from "react-test-renderer";
import LandingPage from "../../src/pages/landingPage/landingPage";

describe('LandingPage', () => {
  it('renders', () => {
    //Arrange
    const component = renderer
      .create(<LandingPage/>)
      .toJSON()

    //Act


    //Assess
    expect(component).toMatchSnapshot()
  })
})