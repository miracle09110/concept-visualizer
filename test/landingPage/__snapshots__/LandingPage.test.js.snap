// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage renders 1`] = `
[
  <nav>
    <h1>
      Concept Visualizer
    </h1>
    <div
      className="hamburger_menu"
    >
      =
    </div>
  </nav>,
  <div
    className="header_section"
  >
    <div
      className="text_container"
    >
      <h1>
        Concept Visualizers: Explore Algorithms, React, and Web Dev Concepts
      </h1>
      <h3>
        Interactive Tools for Understanding Complex Algorithms, ReactJS and Web Development Concepts
      </h3>
      <a
        href="#explore"
      >
        <button>
          Explore!
        </button>
      </a>
    </div>
  </div>,
  <div
    className="explore_container"
    id="explore"
  >
    <h2>
      Explore
    </h2>
    <div
      className="card"
    >
      <h3
        className="main_text"
      >
        Algorithms
      </h3>
      <div
        className="item_container"
      >
        <div
          className="item"
        >
          <a>
            Bubble Sort
          </a>
          <div
            className="placeholder"
          />
        </div>
        <div
          className="item"
        >
          <a>
            Selection Sort
          </a>
          <div
            className="placeholder"
          />
        </div>
        <div
          className="item"
        >
          <a>
            Insertion Sort
          </a>
          <div
            className="placeholder"
          />
        </div>
        <div
          className="item"
        >
          <a>
            Quick Sort
          </a>
          <div
            className="placeholder"
          />
        </div>
        <div
          className="item"
        >
          <a>
            Merge Sort
          </a>
          <div
            className="placeholder"
          />
        </div>
        <div
          className="item"
        >
          <a>
            Linear Search
          </a>
          <div
            className="placeholder"
          />
        </div>
        <div
          className="item"
        >
          <a>
            Depth first
          </a>
          <div
            className="placeholder"
          />
        </div>
        <div
          className="item"
        >
          <a>
            Width first
          </a>
          <div
            className="placeholder"
          />
        </div>
        <div
          className="item"
        >
          <a>
            Jump Search
          </a>
          <div
            className="placeholder"
          />
        </div>
        <div
          className="item"
        >
          <a>
            Binary Search
          </a>
          <div
            className="placeholder"
          />
        </div>
        <div
          className="item"
        >
          <a>
            Ternary Search
          </a>
          <div
            className="placeholder"
          />
        </div>
        <div
          className="item"
        >
          <a>
            Exponential Search
          </a>
          <div
            className="placeholder"
          />
        </div>
        <div
          className="item"
        >
          <a>
            Cycle detection
          </a>
          <div
            className="placeholder"
          />
        </div>
        <div
          className="item"
        >
          <a>
            Dijkstraâ€™s algorithm
          </a>
          <div
            className="placeholder"
          />
        </div>
        <div
          className="item"
        >
          <a>
            A* algorithm
          </a>
          <div
            className="placeholder"
          />
        </div>
      </div>
    </div>
    <div
      className="card"
    >
      <h3
        className="main_text"
      >
        React Concepts
      </h3>
      <div
        className="item_container"
      >
        <div
          className="item"
        >
          <a>
            State and Props
          </a>
          <div
            className="placeholder"
          />
        </div>
        <div
          className="item"
        >
          <a>
            useContext
          </a>
          <div
            className="placeholder"
          />
        </div>
      </div>
    </div>
    <div
      className="card"
    >
      <h3
        className="main_text"
      >
        Web Concepts
      </h3>
      <div
        className="item_container"
      >
        <div
          className="item"
        >
          <a>
            Server-Client
          </a>
          <div
            className="placeholder"
          />
        </div>
        <div
          className="item"
        >
          <a>
            REST API
          </a>
          <div
            className="placeholder"
          />
        </div>
        <div
          className="item"
        >
          <a>
            Webhook
          </a>
          <div
            className="placeholder"
          />
        </div>
      </div>
    </div>
  </div>,
]
`;
